### ISSUE TITLE
Canada: Observed New Year's day for 2021/2022 incorrect

### ISSUE THREAD

**Issue by jasonjensen** [dr-prodigy/python-holidays#707]
When querying December 31st, 2021 in Ontario, Canada, it is listed as "New Year's Day (Observed)", but this is not the case, it was observed on the following Monday. (see for example here: https://www.timeanddate.com/calendar/?year=2022&country=27 , and here https://www.timeanddate.com/holidays/canada/new-year-day)

This doesn't seem to be a problem with the data but with the logic of when the observed day falls. This seems related to these issues/PRs:
https://github.com/dr-prodigy/python-holidays/pull/579
https://github.com/dr-prodigy/python-holidays/issues/581

Thank you for maintaining this package.

```
import holidays
from datetime import datetime
h = holidays.country_holidays("CA", subdiv="ON")
h.get(datetime(2021,12,31))
```

**Comment by KJhellico** [comment]
Seems to be fixed in #811

**Comment by dr-prodigy** [comment]
Thank you @KJhellico , it seems correct to me as well.. can you confirm @jasonjensen ? Thx

**Comment by jasonjensen** [comment]
> Thank you @KJhellico , it seems correct to me as well.. can you confirm @jasonjensen ? Thx

Yes, thank you @dr-prodigy and @KJhellico , this was the one and same issue. This can be closed once #811 is incorporated into the main branch, or now if that is your convention.

**Comment by dr-prodigy** [comment]
Yes, we usually close issues upon their release on beta branch. Thank you @jasonjensen !
